File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int[] result = new int[nums1.length];
        Stack<Integer> st = new Stack<>();
        HashMap<Integer,Integer> map = new HashMap<>();
        
        for(int num:nums2){
            while(!st.empty() && num > st.peek()){
                map.put(st.pop(),num);
            }
            st.push(num);
        }
        int i=0;
        for(int num:nums1){
            result[i++] = map.getOrDefault(num,-1);
        }
        return result;
    }
}

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        for target in nums1:
            ng = -1
            target_found = False
            for num in nums2:
                if num == target:
                    target_found= True
                elif target_found:
                    if num > target:
                        ng = num
                        break
            res.append(ng)
        return res